ios/brave-ios/Sources/PlaylistUI/PlaylistSidebarList.swift

```diff
@@ -0,0 +1,325 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import Data
+import Foundation
+import Playlist
+import SwiftUI
+
+struct PlaylistSidebarList: View {
+  var folders: [PlaylistFolder]
+  var selectedFolderID: PlaylistFolder.ID
+  @Binding var selectedItemID: PlaylistItem.ID?
+  var isPlaying: Bool
+
+  @Environment(\.openTabURL) private var openTabURL
+  @FetchRequest private var items: FetchedResults<PlaylistItem>
+  private typealias PlaylistItemUUID = String
+  @State private var downloadStates: [PlaylistItemUUID: PlaylistDownloadManager.DownloadState] = [:]
+  @State private var downloadProgress: [PlaylistItemUUID: Double] = [:]
+
+  init(
+    folders: [PlaylistFolder],
+    folderID: PlaylistFolder.ID,
+    selectedItemID: Binding<PlaylistItem.ID?>,
+    isPlaying: Bool
+  ) {
+    self.folders = folders
+    self.selectedFolderID = folderID
+    self._items = FetchRequest<PlaylistItem>(
+      sortDescriptors: [
+        .init(keyPath: \PlaylistItem.order, ascending: true),
+        .init(keyPath: \PlaylistItem.dateAdded, ascending: false),
+      ],
+      predicate: .init(format: "playlistFolder.uuid == %@", folderID)
```

Comment:
Hmm, maybe we can make an extension for this FetchRequest? I've seen it a few times I think?

Reply:
Im not actually sure how that would work since its a property wrapper 🤔 maybe a new convenience initializer with a new type that I can make playlist specific ones for?